### PR TITLE
Fleet-addon capd tests

### LINKED ISSUE #64: Fleet-addon automation for capd
support for fleet-addon - implementation will be done only for capd clusterclass cluster which doesn't have CNI installed by default (currently done over CRS resource).

### PR DESCRIPTION
**Pull request by thehejik**
fixes #64 

### Checklist:
- [x] Squashed commits into logical changes
- [ ] Documentation
- [x] GitHub Actions (if applicable): 
https://github.com/rancher/rancher-turtles-e2e/actions/runs/10722714276 (`@short`)
https://github.com/rancher/rancher-turtles-e2e/actions/runs/10733606451 (`@short @full`)

### CODE REVIEW

File: tests/cypress/latest/e2e/unit_tests/capd_kubeadm_cluster.spec.ts [hunk 1]
```diff
@@ -24,7 +24,7 @@ describe('Import CAPD Kubeadm', { tags: '@short' }, () => {
   const repoUrl = "https://github.com/rancher/rancher-turtles-e2e.git"
   const basePath = "/tests/assets/rancher-turtles-fleet-example/"
   const pathNames = ['namespace_autoimport', 'cluster_autoimport', 'clusterclass_autoimport']
-  const branch = "main"
+  const branch = "fleet-addon-capd" // CHANGE ME TO "main" before merging
```
Comment (thehejik) [inline tests/cypress/latest/e2e/unit_tests/capd_kubeadm_cluster.spec.ts:27]: This must be changed to `main` before merging (needed for testing).

File: tests/assets/rancher-turtles-fleet-example/clusterclass_autoimport/clusters/cluster1.yaml [hunk 2]
```diff
@@ -19,7 +19,7 @@ spec:
     class: quick-start
     controlPlane:
       metadata: {}
-      replicas: 3
+      replicas: 1
```
Comment (thehejik) [inline tests/assets/rancher-turtles-fleet-example/clusterclass_autoimport/clusters/cluster1.yaml:22]: We don't test scaling for clusterclass so 1 should be ok

File: tests/assets/rancher-turtles-fleet-example/clusterclass_autoimport/clusters/cluster1.yaml [hunk 3]
```diff
@@ -30,12 +30,12 @@ spec:
     - name: podSecurityStandard
       value:
         audit: restricted
-        enabled: true
+        enabled: false
```
Comment (thehejik) [inline tests/assets/rancher-turtles-fleet-example/clusterclass_autoimport/clusters/cluster1.yaml:33]: This is necessary for `hostNetwork: true` of fleet-agent pod running in ds cluster before any CNI is installed.

File: tests/cypress/latest/support/commands.ts [hunk 4]
```diff
@@ -215,6 +235,65 @@ Cypress.Commands.add('installApp', (appName, namespace) => {
   cy.namespaceReset();
 });
 
+// Command for patching generic YAML resources
+Cypress.Commands.add('patchYamlResource', (clusterName, namespace, resourceKind, resourceName, patch) => {
```
Comment (thehejik) [inline tests/cypress/latest/support/commands.ts:239]: This is crazy command but can be used for patching almost any YAML resource.

File: tests/cypress/latest/e2e/unit_tests/providers_setup.spec.ts [hunk 5]
```diff
@@ -68,7 +68,46 @@ describe('Enable CAPI Providers', () => {
...
+
+    it('Custom Fleet addon config', () => {
+      // Allows Fleet addon to be installed on specific clusters only
+      // Enables hostNetwork for Fleet addon
+
+      const clusterName = 'local';
+      const resourceKind = 'configMap';
+      const resourceName = 'fleet-addon-config';
+      const namespace = 'rancher-turtles-system';
+      const patch = {
```
Comment (cpinjani) [inline tests/cypress/latest/e2e/unit_tests/providers_setup.spec.ts:80]: Can be changed to one-liner for better readability ?
// Patch example: const patch = {data: {manifests: {isNestedIn: true, spec: {...}}};

File: tests/cypress/latest/support/commands.ts [hunk 6]
```diff
@@ -194,14 +196,32 @@ Cypress.Commands.add('addCloudCredsGCP', (name, gcpCredentials) => {
...
   cy.clickButton('Install');
   cy.contains('.outer-container > .header', appName);
   cy.clickButton('Next');
+
+  if (questions != undefined) {
+    // Needs to be adapted for different Apps with questions
+    if (appName == 'Rancher Turtles') {
+      cy.contains('Customize install settings').should('be.visible').click();
+    }
+
```
Comment (cpinjani) [inline tests/cypress/latest/support/commands.ts:213]: Please add documentation for usage of below code

File: tests/assets/rancher-turtles-fleet-example/clusterclass_autoimport/clusters/cluster1.yaml [hunk 7]
```diff
@@ -4,7 +4,7 @@ metadata:
   name: cluster1
   namespace: default
   labels:
-    cni: quick-start-crs-0
+    cni: by-fleet-addon
```
Comment (cpinjani) [inline tests/assets/rancher-turtles-fleet-example/clusterclass_autoimport/clusters/cluster1.yaml:7]:
```suggestion
    cni: by-fleet-addon-kindnet
```

**Comment by thehejik** [comment]
Testrun after addressing comments with `@short` https://github.com/rancher/rancher-turtles-e2e/actions/runs/10772722123

**Review by cpinjani — APPROVED** [review]

**Pull request merged**